ilastik/applets/dataSelection/dataSelectionSerializer.py

```diff
@@ -212,6 +218,37 @@ def importStackAsLocalDataset(self, info, sequence_axis='t'):
             opLoader.SequenceAxis.setValue(sequence_axis)
             opLoader.GlobString.setValue(globstring)
             data_slot = opLoader.Output
+        elif firstPathParts.extension.lower() in OpStreamingHdf5SequenceReaderM.H5EXTS:
+            # Now use the .checkGlobString method of the stack readers
+            isSingleFile = True
+            try:
+                OpStreamingHdf5SequenceReaderS.checkGlobString(globstring)
+            except (OpStreamingHdf5SequenceReaderS.NoInternalPlaceholderError,
+                    OpStreamingHdf5SequenceReaderS.NotTheSameFileError,
+                    OpStreamingHdf5SequenceReaderS.ExternalPlaceholderError):
+                isSingleFile = False
+
+            isMultiFile = True
+            try:
+                OpStreamingHdf5SequenceReaderM.checkGlobString(globstring)
+            except (OpStreamingHdf5SequenceReaderM.NoExternalPlaceholderError,
+                    OpStreamingHdf5SequenceReaderM.SameFileError,
+                    OpStreamingHdf5SequenceReaderM.InternalPlaceholderError):
+                isMultiFile = False
+
+            assert (not(isMultiFile and isSingleFile)), (
+                "Something is wrong, glob string shouldn't allow both")
+            assert (isMultiFile or isSingleFile), (
+                "Glob string doesn't conform to h5 stack glob string rules")
+
+            if isSingleFile:
+                opLoader = OpStreamingHdf5SequenceReaderS(parent=self.topLevelOperator.parent)
```

Comment:
I have addressed this issue by changing the interface of `OpStreamingHdf5SequenceReaderS` in lazyflow. The hdf5 file is now handled in this class and derived from the `GlobString`. The attribute `Hdf5File` was, therefore, removed.